Central-Server/API/internal/jobs.py

```diff
@@ -0,0 +1,51 @@
+"""This file exposes classes, enums, and interfaces for storing data in relation to jobs."""
+import enum
+
+from apiflask import Schema
+from apiflask.fields import Integer, String, DateTime
+from apiflask.validators import Length, OneOf, Regexp
+
+import internal.sanitizers as sanitizers
+
+class JobStatus(enum.Enum):
+    """An enum describing the current status of a job"""
+    QUEUED = 0
+    """Job is queued and will be run on the next available TARS"""
+    RUNNING = 1
+    """Job is currently running on TARS"""
+    CANCELED = 2
+    """Job was canceled by someone `#cancelculture`"""
+    SUCCESS = 3
+    """Job was successfully run"""
+    FAILED_CRASHED = 4
+    """Job crashed on TARS"""
+    FAILED_COMPILE_ERROR = 5
+    """Job failed to compile"""
+    FAILED_TIMEOUT = 6
+    """Job timed out, cancelled `#cancelculturestrikesagain`"""
+    FAILED_OTHER = 7
+    """Job failed for some other reason"""
+    SETUP = 8
+    """Job is currently in the process of being set up."""
+
+class JobRequestSchema(Schema):
+    """The database schema for a generic job request"""
+    commit = String(required=True, validate=Regexp(sanitizers.git_hash_regexp()))
+    username = String(required=True, validate=Regexp(sanitizers.github_username_regexp()))
+    branch = String(required=True, validate=Regexp(sanitizers.branch_name_regexp()))
+    description = String(required=False, load_default="", dump_default="")
+    data_uri = String(required=False, load_default="/api/temp/data", dump_default="/api/temp/data")
+
+# do keep this updated with the db please
+class JobOutSchema(Schema):
```

Comment:
Is there a way to automatically keep this updated?

Reply:
Not really, we could write a SQL autoupdater but that will be a bigger project